src/mon/NVMeofGwMon.cc

```diff
@@ -0,0 +1,530 @@
+            }
+            else if(pending_map.created_gws[group_key][gw_id].performed_full_startup == false){
+                pending_map.created_gws[group_key][gw_id].performed_full_startup = true;
+                propose = true;
+                goto set_propose;
+            }
+        }
+    }
+    else { // gw already created
+        if (gw != group_gws.end()) // if GW reports Available but in monitor's database it is Unavailable
+                                   // it means it did not perform "exit" after failover was set by NVMeofGWMon
+           if( pending_map.created_gws[group_key][gw_id].availability == gw_availability_t::GW_UNAVAILABLE  &&
+               pending_map.created_gws[group_key][gw_id].performed_full_startup == false &&
+               avail == gw_availability_t::GW_AVAILABLE ) {
+               ack_map.created_gws[group_key][gw_id] = pending_map.created_gws[group_key][gw_id];
+               ack_map.epoch = map.epoch;
+               dout(1) << " Force gw to exit: Sending ack_map to GW: " << gw_id << dendl;
+               auto msg = make_message<MNVMeofGwMap>(ack_map);
+               mon.send_reply(op, msg.detach());
+               goto false_return;
+           }
+    }
+
+    // At this stage the gw has to be in the Created_gws
+    if(gw == group_gws.end()){
+        dout(1) << "Error : Administratively deleted GW sends beacon " << gw_id <<dendl;
+        goto false_return; // not sending ack to this beacon
+    }
+
+    // deep copy the whole nonce map of this GW
+    if(m->get_nonce_map().size()) {
+        if(pending_map.created_gws[group_key][gw_id].nonce_map != m->get_nonce_map())
+        {
+            dout(10) << "nonce map of GW  changed , propose pending " << gw_id << dendl;
+            pending_map.created_gws[group_key][gw_id].nonce_map = m->get_nonce_map();
+            dout(10) << "nonce map of GW " << gw_id << " "<< pending_map.created_gws[group_key][gw_id].nonce_map  << dendl;
+            nonce_propose = true;
+        }
+    }
+    else  {
+        dout(10) << "Warning: received empty nonce map in the beacon of GW " << gw_id << " "<< dendl;
+    }
+
+    if(sub.size() == 0 ) {
+        avail = gw_availability_t::GW_UNAVAILABLE;
+    }
+    if(pending_map.created_gws[group_key][gw_id].subsystems != sub)
+    {
+        dout(10) << "subsystems of GW changed, propose pending " << gw_id << dendl;
+        pending_map.created_gws[group_key][gw_id].subsystems =  sub;
+        dout(20) << "subsystems of GW " << gw_id << " "<< pending_map.created_gws[group_key][gw_id].subsystems << dendl;
+        nonce_propose = true;
+    }
+    pending_map.created_gws[group_key][gw_id].last_gw_map_epoch_valid = ( map.epoch == m->get_last_gwmap_epoch() );
+    if( pending_map.created_gws[group_key][gw_id].last_gw_map_epoch_valid == false ){
+      dout(1) <<  "map epoch of gw is not up-to-date " << gw_id << " epoch " << map.epoch << " beacon_epoch " << m->get_last_gwmap_epoch() <<  dendl;
```

Comment:
Why would a gateway being behind on maps be a monitor error requiring logging? Being behind on a map is a perfectly normal state of affairs.